commit 2ffb3518eb89cf003085b2700b40d83768630c8a (HEAD -> master)
Author: Tuetuopay <tuetuopay@me.com>
Date:   Mon Mar 15 01:51:15 2021 +0100

    input: Add -nokb flag to disable keyboard input
    
    This is useful to run this over a serial port

diff --git a/fbdoom/i_input_tty.c b/fbdoom/i_input_tty.c
index e66e17b..04ececc 100644
--- a/fbdoom/i_input_tty.c
+++ b/fbdoom/i_input_tty.c
@@ -53,6 +53,9 @@ int vanilla_keyboard_mapping = 1;
 
 static int shiftdown = 0;
 
+// Should we take no keyboard input?
+static int no_kb = 0;
+
 // Lookup table for mapping AT keycodes to their doom keycode
 static const char at_to_doom[] =
 {
@@ -420,6 +423,9 @@ static void UpdateShiftStatus(int pressed, unsigned char key)
 
 void I_GetEvent(void)
 {
+    if (no_kb)
+        return;
+
     event_t event;
     int pressed;
     unsigned char key;
@@ -485,6 +491,10 @@ void I_GetEvent(void)
 
 void I_InitInput(void)
 {
+    no_kb = M_CheckParm("-nokb");
+    if (no_kb)
+        return;
+
     kbd_init();
 
     //UpdateFocus();
